### ISSUE TITLE
txnbuild: generalise "Price" field

### ISSUE THREAD

**Issue by ire-and-curses** [stellar/go#1028]
Currently, `txnbuild.ManageOffer.Price` is a string. It would be cool to support number, or numerator/denominator (JS SDK supports [multiple types](https://stellar.github.io/js-stellar-sdk/Operation.html#.manageOffer)). Existing package [amount](https://github.com/stellar/go/blob/master/amount/main.go) has overlap with this and could potentially be used/extended.

**Comment by poliha** [comment]
Here, I propose we make `ManageOffer.Price` an `Interface{}`. This will enable it support `string`, `float` and horizon protocol `Price` types. 
We can validate the types when `BuildXDR()` is called and return an error appropriately.

**Comment by ire-and-curses** [comment]
@poliha could you expand on why this is a breaking change? Naively I imagined we could do something like this:

```
func parsePrice(p interface{}) Price {
    switch p := p.(type) {
    case string:
    case float64:
    ...
```

I guess if we store the resulting price as something other than a string (e.g. a new `Price` object that can handle being instantiated from any of these types) then I agree it's a breaking change. Is that what you were thinking?

**Comment by poliha** [comment]
@ire-and-curses Yes that is what I had in mind. Currently, users expect `Price` to be a string. To generalise it, I was thinking the structs will become something like this 
```golang
type ManageSellOffer struct {
	Selling       Asset
	Price         interface{}
	...
}
```

then we can call a function like `parsePrice` above when building or validating the struct fields.
The type change for `Price` field is what i referred to as the breaking change.
